mcstatus/status_response.py

```diff
@@ -0,0 +1,640 @@
+    """Class which extends a `StatusPlayers` class with list of players.
+
+    :param players: List of players. Can be empty even if online > 0.
+    """
+
+    list: List[JavaStatusPlayer]
+
+    @classmethod
+    def build(cls, raw: Dict[str, Any]) -> Self:
+        """Build `JavaStatusPlayers` from raw response dict.
+
+        :param raw: Raw response dict.
+        :raise ValueError: If the required keys (online, max) are not present.
+        :raise TypeError: If the required keys (online - int, max - int, sample - list) are not of the specified type.
+        :return: `JavaStatusPlayers` object.
+        """
+        _validate_data(raw, "players", [("online", int), ("max", int)])
+        if "sample" in raw:
+            _validate_data(raw, "players", [("sample", list)])
+        return cls(
+            online=raw["online"],
+            max=raw["max"],
+            list=[JavaStatusPlayer.build(player) for player in raw["sample"]] if "sample" in raw else [],
+        )
+
+
+@dataclass
+class BedrockStatusPlayers(StatusPlayers):
+    """Class which represents `BedrockStatusResponse` players field."""
+
+
+@dataclass
+class JavaStatusPlayer:
+    """Class for storing player info. Only for Java."""
+
+    name: str
+    uuid: str
+
+    @classmethod
+    def build(cls, raw: Dict[str, Any]) -> Self:
+        """Build `JavaStatusPlayer` from raw response dict.
+
+        :param raw: Raw response dict.
+        :raise ValueError: If the required keys (name, uuid) are not present.
+        :raise TypeError: If the required keys (name - str, uuid - str) are not of the specified type.
+        :return: `JavaStatusPlayer` object.
+        """
+        _validate_data(raw, "player", [("name", str), ("id", str)])
+        return cls(name=raw["name"], uuid=raw["id"])
+
+
+@dataclass
+class StatusVersion(AbstractDataclass):
+    """Class for storing version info.
+
+    :param name: Version name. Example `1.18.0`.
+    :param protocol: Version protocol. See https://minecraft.fandom.com/wiki/Protocol_version.
+    """
+
+    name: str
+    protocol: int
+
+
+@dataclass
+class JavaStatusVersion(StatusVersion):
+    """Class for storing Java version info."""
+
+    @classmethod
+    def build(cls, raw: Dict[str, Any]) -> Self:
+        """Build `JavaStatusVersion` from raw response dict.
+
+        :param raw: Raw response dict.
+        :raise ValueError: If the required keys (name, protocol) are not present.
+        :raise TypeError: If the required keys (name - str, protocol - int) are not of the specified type.
+        :return: `JavaStatusVersion` object.
+        """
+        _validate_data(raw, "version", [("name", str), ("protocol", int)])
+        return cls(name=raw["name"], protocol=raw["protocol"])
+
+
+@dataclass
+class BedrockStatusVersion(StatusVersion):
+    """Class for storing Bedrock version info.
+
+    :param brand: Like `MCPE` or another.
+    """
+
+    brand: str
+
+
+class JavaStatusResponse(NewJavaStatusResponse):
```

Comment:
We may want to define these deprecated classes in `pinger.py` or `bedrock_status.py` respectively. There's no real reason why they would need to be in here, and it defining them in the files they were defined in previously makes a bit more sense and could be a bit easier to later find and deprecate.

Reply:
I answered it [here](https://github.com/py-mine/mcstatus/pull/306#discussion_r872993704).